### DISCUSSION TITLE
Can not SSR styles on Next.js  when using createStyles

### DISCUSSION THREAD

**Discussion by liringlas** [mantinedev/mantine#2406] [Q&A] [answered]
Good day!
I'm using `createStyles` for CSS in my Next.js app, but I could not achieve server side rendering of those styles.
I built everything using `next build` with `getStaticProps` for pages, but had no success.
Then I also tried example with explicit usage of `createStylesServer` from here:
https://mantine.dev/guides/next/#create-your-own-getinitialprops
But the result is the same.

Could you please clarify if it is even possible to render pages and components with SSR styles from `createStyles`?
And if yes, what exactly I am missing?
I'm not using custom emotion cache, only `createStyles` itself.
Thanks in advance.

**Comment by rtivital** [comment, marked as answer]
Hi @liringlas, you can use one of the templates to get started, all of them have ssr preconfigured to work correctly.

**Reply by liringlas** [reply]
I've been digging https://github.com/mantinedev/mantine-next-template for the past 40 minutes since your reply.
I can see that it clearly has working SSR styles for createStyles.
Tried to copy `_document` and `_app` from there to almost identical state -- but my app just refuses to produce SSR for createStyles.
Sigh. Thank you anyway, I'll keep digging.

**Reply by rtivital** [reply]
Try using yarn to install dependencies

**Reply by liringlas** [reply]
I do. And I have those dependencies as in https://github.com/mantinedev/mantine-next-template
I understand that it's not really informative, so maybe if I get a reproduction repo tomorrow -- I'll share it.

**Reply by liringlas** [reply]
Well. Latest update is that I removed `node_modules` and `yarn.lock`, then ran `yarn` again.
After it all was complete I tried once more -- and now SSR for `createStyles` works just as in your example repo.
Magic it is, sir :D
Thanks for communication.
